tooling/cli/lib/deploy.js

```diff
@@ -334,3 +342,11 @@ function wait(ms = 1000) {
     setTimeout(resolve, ms);
   });
 }
+
+function getWorkerVersion() {
+  const { vite: viteVersion } = packageManifest.dependencies || {};
+  if (semver.satisfies(semver.coerce(viteVersion), '>=3')) {
+    return 'v2';
+  }
+  return 'v1';
+}
```

Comment:
determine worker version based on the `vite` version specified in package.json file. there might be a better way to check for the worker version.